### PR TITLE
Fixed az sql mi-arc show for indirect scenario

### PR DESCRIPTION
**Pull request by candiceye**
The output of 'az sql mi-arc show' has a different shape for indirect vs. direct mode. Right now, we only cover direct mode shape. This PR takes the output of **either** the direct mode shape or the indirect mode shape, and combines it into one "normalized" shape called SqlMiShowResult.

### CODE REVIEW

File: extensions/azcli/src/typings/az-ext.d.ts [hunk 1]
```diff
@@ -137,53 +137,89 @@ declare module 'az-ext' {
...
+				},
+				logs: {
+					volumes: StorageVolume[]
+				}
+			}
+		},
+		status: {
+			readyReplicas: string, // "1/1"
+			state: string, // "Ready",
+			logSearchDashboard: string, // https://127.0.0.1:30777/kibana/app/kibana#/discover?_a=(query:(language:kuery,query:'custom_resource_name:miaa1'))
```
Comment (swells) [inline extensions/azcli/src/typings/az-ext.d.ts:169]: these comments `//` are just notes on the shape of what will occupy these props and not dead comments?

Comment (Charles-Gagnon) [reply extensions/azcli/src/typings/az-ext.d.ts:169]: Correct (or at least that's what I started them out as 😄 )

Comment (candiceye) [reply extensions/azcli/src/typings/az-ext.d.ts:169]: Yep, just examples of what it may look like

File: extensions/azcli/src/az.ts [hunk 2]
```diff
@@ -301,6 +303,63 @@ export class AzTool implements azExt.IAzApi {
...
+
+			let stdout = <unknown>result.stdout;
+			let stderr = <unknown>result.stderr;
+
+			try {
+				// Automatically try parsing the JSON. This is expected to fail for some az commands such as resource delete.
+				stdout = JSON.parse(result.stdout);
+			} catch (err) {
+				// If the output was not pure JSON, catch the error and log it here.
+				Logger.log(loc.azOutputParseErrorCaught(args.concat(['--output', 'json']).toString()));
```
Comment (Charles-Gagnon) [inline extensions/azcli/src/az.ts:323]: Do you want to continue on at this point? It'll likely fall into the else and then try to access stdout.metadata.name which will likely fail

**Review by Charles-Gagnon — APPROVED** [review]

File: extensions/azcli/src/az.ts (hunk 2 above)
Comment (candiceye) [reply extensions/azcli/src/az.ts:323]: Good point, I'll prevent the func from continuing here

**Pull request merged**